pymc3/tests/test_sampling.py

```diff
@@ -285,6 +285,27 @@ def callback(trace, draw):
             assert len(trace) == trace_cancel_length
 
 
+def test_sample_find_MAP_does_not_modify_start():
+    # see https://github.com/pymc-devs/pymc3/pull/4458
+    with pm.Model():
+        pm.Lognormal("untransformed")
+
+        # make sure find_Map does not modify the start dict
+        start = {"untransformed": 2}
+        pm.find_MAP(start=start)
+        assert start == {"untransformed": 2}
+
+        # make sure sample does not modify the start dict
+        start = {"untransformed": 0.2}
+        pm.sample(draws=10, step=pm.Metropolis(), tune=5, start=start, chains=3)
```

Comment:
Looks very good! Small request, can you change the number of chains to 2 here as well? It makes it more obvious that there is nothing special about the number of chains.

Reply:
Actually nvm